src/common/events/EventController.cpp

```diff
@@ -0,0 +1,43 @@
+#include "EventController.hpp"
+
+#include <iostream>
+
+
+EventController::EventController() {}
+
+void EventController::addEventListener(EventType event_type, EventCallback event_callback) {
+    mEventMapLock.lock();
+    mEventCallbackMap[event_type].push_back(event_callback);
+    mEventMapLock.unlock();
+}
+
+void EventController::queueEvent(std::shared_ptr<Event> event) {
+    mEventQueueLock.lock();
+    mEventQueue.push(event);
+    mEventQueueLock.unlock();
+}
+
+void EventController::forceProcessEvents() {
```

Comment:
shouldn't this function take locks since it is modifying mEventQueue?

Reply:
Hence the "force" prefix. 
This means that the thread that it's up to the thread calling this function to make sure the used variables are thread safe to write/read from.

It's designed this way so that GameController (the master thread) can unlock/lock `EventController` in `collectClientInputs`, which is not the same place where `processEvents` is called.